src/lib.rs

```diff
@@ -1960,3 +1991,72 @@ impl<T, L: CapacityLimiter<T>> Drop for InactiveReceiver<T, L> {
         }
     }
 }
+
+/// A [`CapacityLimiter`] based on the memory size of simple objects like [`Box`] or [`Vec`].
+///
+/// This counts the size of the element plus the size of the value it Derefs to.  This is only
+/// correct of there is exactly one heap allocation and that allocation is returned by the deref
+/// operator.  This means it does not account for the unused capacity of a `Vec`, for other heap
+/// pointers contained within a boxed type, or for the refcounts of an [`Arc`].
+#[derive(Copy, Clone, Debug)]
+pub struct BoxlikeCapacityLimit {
```

Comment:
`BoxlikeCapacityLimiter` for consistency? I'd actually just name it `ContainerCapacityLimiter` or `SimpleCapacityLimiter`.